cli/flags/shared/shared.go

```diff
@@ -147,47 +154,91 @@ func NewQueueFlags(opts *options.TerragruntOptions, prefix flags.Prefix) cli.Fla
 				Name:        QueueExcludesFileFlagName,
 				EnvVars:     tgPrefix.EnvVars(QueueExcludesFileFlagName),
 				Destination: &opts.ExcludesFile,
+				Hidden:      true,
 				Usage:       "Path to a file with a list of directories that need to be excluded when running *-all commands.",
 			},
 			flags.WithDeprecatedEnvVars(terragruntPrefix.EnvVars("excludes-file"), terragruntPrefixControl),
 		),
 
 		flags.NewFlag(
 			&cli.SliceFlag[string]{
-				Name:        QueueExcludeDirFlagName,
-				EnvVars:     tgPrefix.EnvVars(QueueExcludeDirFlagName),
-				Destination: &opts.ExcludeDirs,
-				Usage:       "Unix-style glob of directories to exclude from the queue of Units to run.",
+				Name:    QueueExcludeDirFlagName,
+				EnvVars: tgPrefix.EnvVars(QueueExcludeDirFlagName),
+				Hidden:  true,
+				Usage:   "Unix-style glob of directories to exclude from the queue of Units to run.",
+				Action: func(_ *cli.Context, value []string) error {
+					if len(value) == 0 {
+						return nil
+					}
+
+					for _, v := range value {
+						// We explicitly wrap the value in curly braces to ensure that it is treated
+						// as a path expression, and not a name filter.
+						opts.FilterQueries = append(opts.FilterQueries, "!{"+v+"}")
+					}
+
+					return nil
+				},
 			},
 			flags.WithDeprecatedEnvVars(terragruntPrefix.EnvVars("exclude-dir"), terragruntPrefixControl),
 		),
 
 		flags.NewFlag(
 			&cli.SliceFlag[string]{
-				Name:        QueueIncludeDirFlagName,
-				EnvVars:     tgPrefix.EnvVars(QueueIncludeDirFlagName),
-				Destination: &opts.IncludeDirs,
-				Usage:       "Unix-style glob of directories to include from the queue of Units to run.",
+				Name:    QueueIncludeDirFlagName,
+				EnvVars: tgPrefix.EnvVars(QueueIncludeDirFlagName),
+				Hidden:  true,
+				Usage:   "Unix-style glob of directories to include from the queue of Units to run.",
+				Action: func(_ *cli.Context, value []string) error {
+					if len(value) == 0 {
+						return nil
+					}
+
+					for _, v := range value {
+						// We explicitly wrap the value in curly braces to ensure that it is treated
+						// as a path expression, and not a name filter.
+						opts.FilterQueries = append(opts.FilterQueries, "{"+v+"}")
```

Comment:
I suspect this may lead to issues if the included unit has dependencies - with `{"+v+"}`, dependencies will be ignored

Reply:
That's an intended breaking change here. Users should use `{./**}...` if they _want_ dependencies to be included.